pangeo_forge_recipes/patterns.py

```diff
@@ -268,3 +270,22 @@ def prune_pattern(fp: FilePattern, nkeep: int = 2) -> FilePattern:
             assert "Should never happen"
 
     return FilePattern(fp.format_function, *new_combine_dims)
+
+
+# Register custom objects with Dask, to ensure memory usage is tracked properly
+
+
+@dask.sizeof.sizeof.register(FilePattern)
```

Comment:
Actually, these should use the bulitin `__sizeof__` protocol since we control these objects.